core/src/main/java/org/apache/gravitino/storage/relational/service/ModelVersionMetaService.java

```diff
@@ -0,0 +1,288 @@
+import org.apache.gravitino.storage.relational.po.ModelPO;
+import org.apache.gravitino.storage.relational.po.ModelVersionAliasRelPO;
+import org.apache.gravitino.storage.relational.po.ModelVersionPO;
+import org.apache.gravitino.storage.relational.utils.ExceptionUtils;
+import org.apache.gravitino.storage.relational.utils.POConverters;
+import org.apache.gravitino.storage.relational.utils.SessionUtils;
+import org.apache.gravitino.utils.NameIdentifierUtil;
+import org.apache.gravitino.utils.NamespaceUtil;
+import org.glassfish.jersey.internal.guava.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ModelVersionMetaService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ModelVersionMetaService.class);
+
+  private static final ModelVersionMetaService INSTANCE = new ModelVersionMetaService();
+
+  public static ModelVersionMetaService getInstance() {
+    return INSTANCE;
+  }
+
+  private ModelVersionMetaService() {}
+
+  public List<ModelVersionEntity> listModelVersionsByNamespace(Namespace ns) {
+    NamespaceUtil.checkModelVersion(ns);
+
+    NameIdentifier modelIdent = NameIdentifier.of(ns.levels());
+    // Will throw a NoSuchEntityException if the model does not exist.
+    ModelEntity modelEntity = ModelMetaService.getInstance().getModelByIdentifier(modelIdent);
+
+    List<ModelVersionPO> modelVersionPOs =
+        SessionUtils.getWithoutCommit(
+            ModelVersionMetaMapper.class,
+            mapper -> mapper.listModelVersionMetasByModelId(modelEntity.id()));
+
+    if (modelVersionPOs.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    // Get the aliases for all the model versions.
+    List<ModelVersionAliasRelPO> aliasRelPOs =
+        SessionUtils.getWithoutCommit(
+            ModelVersionAliasRelMapper.class,
+            mapper -> mapper.selectModelVersionAliasRelByModelId(modelEntity.id()));
+    Multimap<Integer, ModelVersionAliasRelPO> aliasRelPOsByModelVersion =
+        ArrayListMultimap.create();
+    aliasRelPOs.forEach(r -> aliasRelPOsByModelVersion.put(r.getModelVersion(), r));
+
+    return modelVersionPOs.stream()
+        .map(
+            m -> {
+              List<ModelVersionAliasRelPO> versionAliasRelPOs =
+                  Lists.newArrayList(aliasRelPOsByModelVersion.get(m.getModelVersion()));
+              return POConverters.fromModelVersionPO(modelIdent, m, versionAliasRelPOs);
+            })
+        .collect(Collectors.toList());
+  }
+
+  public ModelVersionEntity getModelVersionByIdentifier(NameIdentifier ident) {
+    NameIdentifierUtil.checkModelVersion(ident);
+
+    NameIdentifier modelIdent = NameIdentifier.of(ident.namespace().levels());
+    // Will throw a NoSuchEntityException if the model does not exist.
+    ModelEntity modelEntity = ModelMetaService.getInstance().getModelByIdentifier(modelIdent);
+
+    Integer modelVersion;
+    try {
+      modelVersion = Integer.valueOf(ident.name());
```

Comment:
What situation the `ident.name()` will be a Integer, and why use this as the model version property? Is the identity like `metalake.catalog.schema.model.1`?

Reply:
Yes, when get a model version using model identifier and version number, I will convert them as a model  version name identifier, using version as a name.

Reply:
My concern is whether this design is universal for other types of resources? Or can we accept that in the future, if we want to access the sub-resources of a certain type of resource, we need to obtain it through the identifier method?

Reply:
It is mainly due to the limitation of the current entity store interface, which only accepts the NameIdentifier as the key, so we have to convert the resource description into a name identifier. Here it is a little tricky, but seems there's no better way unless we refactor the entity store interface.

You can also check the current column's name identifier implementation, it also has 5 levels, so I think it is a feasible way.